caffe2/operators/utility_ops.cu

```diff
@@ -181,8 +181,101 @@ bool GatherOp<CUDAContext>::DoRunWithType() {
   return true;
 }
 
-namespace {
 REGISTER_CUDA_OPERATOR(Gather, GatherOp<CUDAContext>);
+
+/**
+ * @brief Update slices of Y in-place with a batch of weighted X's.
+ * Y[idx] = alpha[b] * X[b][i] + Y[idx]
+ * i=0,...,N-1
+ * b=0,...,B-1
+ * idx=Indices[i]
+ */
+template<typename T_INDEX>
+__global__ void 
+AxpySliceKernel(
+             const TIndex N,
+             const TIndex B,
+             const TIndex slice_size,
+             const float** alpha,
+             const float** X,
+             const T_INDEX* Indices, 
+             float* Y,
+             const TIndex M) {
+  for (int i = blockIdx.x; i < N; i += gridDim.x) {
+    T_INDEX idx = Indices[i];
+    float* y_offset = Y + (idx * slice_size);
+    for (int b = 0; b < B; b++) {
```

Comment:
It's possible to change. Both are correct, however I compared the performance for large sizes and saw no diffs.